bundles/org.openhab.binding.alarmdecoder/src/main/java/org/openhab/binding/alarmdecoder/internal/handler/LRRHandler.java

```diff
@@ -0,0 +1,108 @@
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.smarthome.core.library.types.DecimalType;
+import org.eclipse.smarthome.core.library.types.StringType;
+import org.eclipse.smarthome.core.thing.Bridge;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.alarmdecoder.internal.config.LRRConfig;
+import org.openhab.binding.alarmdecoder.internal.protocol.LRRMessage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link LRRHandler} is responsible for handling long range radio (LRR) messages.
+ *
+ * @author Bob Adair - Initial contribution
+ * @author Bill Forsyth - Initial contribution
+ */
+@NonNullByDefault
+public class LRRHandler extends ADThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(LRRHandler.class);
+
+    private @NonNullByDefault({}) LRRConfig config;
+
+    public LRRHandler(Thing thing) {
+        super(thing);
+    }
+
+    /**
+     * Returns true if this handler is responsible for the supplied partition.
+     * This is true is this handler's partition is 0 (all), the supplied partition is 0 (all), or if this handler's
+     * partition matches the supplied partition.
+     */
+    public Boolean responsibleFor(final int partition) {
+        return (config.partition == partition || config.partition == 0 || partition == 0);
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(LRRConfig.class);
+
+        if (config.partition < 0) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR);
+            return;
+        }
+        logger.debug("LRR handler initializing for partition {}", config.partition);
+
+        initDeviceState();
+
+        logger.trace("LRR handler finished initializing");
+    }
+
+    @Override
+    protected void initDeviceState() {
+        logger.trace("Initializing device state for RLL partition {}", config.partition);
+        Bridge bridge = getBridge();
+        if (bridge == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "No bridge configured");
+        } else if (bridge.getStatus() == ThingStatus.ONLINE) {
+            updateStatus(ThingStatus.ONLINE);
+            initChannelState();
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE);
+        }
+    }
+
+    @Override
+    public void initChannelState() {
+        // Do nothing
+    }
+
+    @Override
+    public void notifyPanelReady() {
+        // Do nothing
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // All channels are read-only, so ignore all commands.
```

Comment:
You still need to handle a RefreshType command if possible.

Reply:
I thought about this, but the AD device mainly just passively listens on the panel's serial bus and passes along the messages it sees. There is no way to query it for state, as I would want to do when receiving a RefreshType command. So I thought it best to just ignore RefreshType along with all other commands.

Reply:
If there is no way to handle a RefreshType then I guess you can just ignore it.

Reply:
Ok. Thanks. I'm open to suggestions, but I can't really thing of an appropriate REFRESH action that the AD would support.